### DISCUSSION TITLE
Werf cleanup question

### DISCUSSION THREAD

**Discussion by bonza** [werf/werf#4696] [General]
Hi,

We have a bit specific issue with our ad-hoc ECR registry cleanup tool and are very interested to understand if werf will help to solve it or not.

We have short-living Kubernetes jobs running as part of HELM deployment using [hooks](https://helm.sh/docs/topics/charts_hooks/), so these jobs are disappearing from Kubernetes right after release is finished. We are using helm manifests to get a list of images that are in use, and apparently, these jobs are not listed there.
```helm get manifest --namespace $NAMESPACE ${CHART_NAME}-${NAMESPACE} > release-manifest.yml```

Does werf use a different approach to get the list of images that are in use?

**Comment by alexey-igrychev** [comment]
We simply take the images of all the resources from Kubernetes (https://github.com/werf/werf/blob/main/pkg/cleaning/allow_list/kubernetes.go).

**Reply by bonza** [reply]
@alexey-igrychev your documentation says: _Created hooks resources will not be deleted after completion unless there is [special annotation "helm.sh/hook-delete-policy": hook-succeeded,hook-failed](https://helm.sh/docs/topics/charts_hooks/)._
So if the resource has this annotation, the resource will be deleted and werf will delete the corresponding image, right?

**Reply by alexey-igrychev** [reply]
werf will not ignore this image when cleaning if it does not find it in Kubernetes, but there are other policies. https://werf.io/documentation/v1.2/advanced/cleanup.html#cleaning-up-outdated-data

**Reply by alexey-igrychev** [reply]
Also, werf cleanup does not keep images for completed and failed jobs (https://github.com/werf/werf/blob/b7edaa3290132a644689c4cd4af4565112976f05/pkg/cleaning/allow_list/kubernetes.go#L220-L231).
